src/attention/utils/helpers.ts

```diff
@@ -1,98 +1,243 @@
+  fallbackPlacements?: Directions[]
+  targetEl?: unknown
+  tooltip?: Boolean
+  popover?: Boolean
+  callout?: Boolean
+  highlight?: Boolean
+  noArrow?: Boolean
+  distance?: number
+  skidding?: number
+  waitForDOM?: () => void
+}
+
+const middlePosition = 'calc(50% - 7px)'
+const isDirectionVertical = (name: string) =>
+  [TOPSTART, TOP, TOPEND, BOTTOMSTART, BOTTOM, BOTTOMEND].includes(name)
+
 function computeCalloutArrow({
   actualDirection,
-  directionName,
+  directionName = 'bottom',
   arrowEl,
 }: AttentionState) {
-  if (!arrowEl) return;
+  if (!arrowEl) return
+
+  actualDirection = directionName
 
-  actualDirection = directionName;
-  const directionIsVertical = isDirectionVertical(directionName);
-  arrowEl.style.left = directionIsVertical ? middlePosition : "";
-  arrowEl.style.top = !directionIsVertical ? middlePosition : "";
+  const arrowDirection = opposites[actualDirection]
+  const arrowRotation = rotation[arrowDirection]
+
+  const directionIsVertical = isDirectionVertical(directionName)
+  arrowEl.style.left = directionIsVertical ? middlePosition : ''
+  arrowEl.style.top = !directionIsVertical ? middlePosition : ''
+  arrowEl.style.transform = `rotate(${arrowRotation}deg)`
 }
 
 export async function useRecompute(state: AttentionState) {
-  if (!state.isShowing) return; // we're not currently showing the element, no reason to recompute
-  await state?.waitForDOM?.(); // wait for DOM to settle before computing
-  if (state.isCallout) return computeCalloutArrow(state); // we don't move the callout box, only its arrow
-  const position = await computePosition(
-    state.targetEl as ReferenceElement,
-    state.attentionEl as HTMLElement,
-    {
-      placement: state.directionName,
-      middleware: [
-        // Should we make this configurable, but have these as sane defaults?
-        flip(),
-        offset(8),
-        shift({ padding: 16 }),
-        // @ts-ignore
-        arrow({ element: state.noArrow ? undefined : state.arrowEl }), // FIXME
-      ],
-    }
-  );
-  // @ts-ignore
-  state.actualDirection = position.placement;
-  Object.assign(state.attentionEl?.style || {}, {
-    left: "0",
-    top: "0",
-    transform: `translate3d(${Math.round(position.x)}px, ${Math.round(
-      position.y
-    )}px, 0)`,
-  });
-  // @ts-ignore
-  let { x, y } = position.middlewareData.arrow;
-
-  if (state.arrowEl) {
-    state.arrowEl.style.left = x ? x + "px" : "";
-    state.arrowEl.style.top = y ? y + "px" : "";
+  if (!state?.isShowing) return // we're not currently showing the element, no reason to recompute
+  if (state?.waitForDOM) {
+    await state?.waitForDOM() // wait for DOM to settle before computing
   }
+  if (state?.isCallout) return computeCalloutArrow(state)
+  
+  const referenceEl: ReferenceElement = state?.targetEl as ReferenceElement
+  const floatingEl: HTMLElement = state?.attentionEl as unknown as HTMLElement
+  const arrowEl: HTMLElement = state?.arrowEl as unknown as HTMLElement
```

Comment:
Is it valid to type something `as unknown as HTMLElement`?

Reply:
I think you can assert it to first `unknown`, but it is considered an unsafe type assertion, so I'm overriding TypeScript by first asserting the variable as unknown and then to the desired type, which in this case is `HTMLElement.` However, I just found this article and saw that is not recommended to use type assertion. It is better to use Type annotation. So I think I will change it to this: 
```
const referenceEl= state?.targetEl
const floatingEl = state?.attentionEl
const arrowEl = state?.arrowEl
```
and then set the types for `targetEl`, `attentionEl` and `arrowEl` in the type for `attentionState` to this: targetEl?: ReferenceElement
attentionEl?: HTMLElement
arrowEl?: HTMLElement

Since when `targetEl`, `attentionEl` and/or `arrowEl`  are defined we want them to have these types and not `unknown` or `null`

Reply:
Updated in commit: [6c52c9d](https://github.com/warp-ds/core/pull/15/commits/6c52c9dd87f76c84b60c6c03389dbcd4302f36c2)